starport/pkg/cosmoscmd/proxy.go

```diff
@@ -0,0 +1,74 @@
+package cosmoscmd
+
+import (
+	"path/filepath"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/cosmosutil"
+	"github.com/tendermint/starport/starport/pkg/ctxticker"
+	"github.com/tendermint/starport/starport/pkg/gitpod"
+	"github.com/tendermint/starport/starport/pkg/xchisel"
+	"github.com/tendermint/starport/starport/services/network/networkchain"
+)
+
+const TunnelRerunDelay = 5 * time.Second
+
+// startProxyForTunneledPeers hooks the `appd start` command to start an HTTP proxy server and HTTP proxy clients for each node that needs HTTP tunneling.
+// HTTP tunneling is activated ** ONLY** if your app's `$APP_HOME/config` dir has an `spn.yml` file and only if this file has `tunneled_peers` field inside with a list of
+// tunneled peers/nodes.
+//
+// If you're using SPN as coordinator and do not want to allow HTTP tunneling feature at all,
+// you can prevent `spn.yml` file to being generated by not approving validator requests
+// that has HTTP tunneling enabled instead of plain TCP connections.
+func startProxyForTunneledPeers(clientCtx client.Context, cmd *cobra.Command) {
+	if cmd.Name() != "start" {
+		return
+	}
+	serverCtx := server.GetServerContextFromCmd(cmd)
+	ctx := cmd.Context()
+
+	spnConfigPath := filepath.Join(clientCtx.HomeDir, cosmosutil.ChainConfigDir, networkchain.SPNConfigFile)
+	spnConfig, err := networkchain.GetSPNConfig(spnConfigPath)
+	if err != nil {
+		serverCtx.Logger.Error("Failed to open spn config file", "reason", err.Error())
+	}
+	for _, peer := range spnConfig.TunneledPeers {
+		if peer.Name == networkchain.HTTPTunnelChisel {
+			peer := peer
+			go func() {
+				ctxticker.DoNow(ctx, TunnelRerunDelay, func() error {
+					serverCtx.Logger.Info("Starting chisel client", "tunnelAddress", peer.Address, "localPort", peer.LocalPort)
+					err := xchisel.StartClient(ctx, peer.Address, peer.LocalPort, "26656")
+					if err != nil {
+						serverCtx.Logger.Error("Failed to start chisel client",
+							"tunnelAddress", peer.Address,
+							"localPort", peer.LocalPort,
+							"reason", err.Error(),
+						)
+					}
+					return nil
+				})
+			}()
+		}
+	}
+
+	if gitpod.IsOnGitpod() && len(spnConfig.TunneledPeers) > 0 {
```

Comment:
```suggestion
	if len(spnConfig.TunneledPeers) > 0 {
```

Reply:
Chisel server i necessary to run only on gitpod. This code will also be running on vm's and without this check chisel server will start on them.

Reply:
Uh, I understand.